app/controllers/users_controller.rb

```diff
@@ -0,0 +1,66 @@
+class UsersController < AuthenticationController
+  skip_before_action :authenticate, only: :create
+  # valid user creation curl --
+  #
+  # curl -X POST -F user[first_name]=Nate \
+  # -F user[state]=MD -F user[country]=USA \
+  # -F user[city]=Annapolis -F user[email]=test@test.com \
+  # -F user[last_name]=Bates  -F user[password]=Helloworld \
+  # http://localhost:3000/users.json -H "Accept: application/json"
+
+  # curl -i -H "Content-Type: application/json" \
+  # -d '{"user": {"email": "test@test.com"}}'
+
+  # POST /users
+  def create
+    result = CreateUser.call(user_params: user_params)
+    if result.success?
+      token = CreateAuthToken.call(user: result.user).token
+      render json: { jwt: token }, status: :created
+    else
+      render json: result.user.errors, status: :unprocessable_entity
+    end
+  end
+
+  # GET /users/1
+  def show
+    result = ShowUser.call(id: params[:id])
+    if result.success?
+      render json: result.user, status: :ok
+    else
+      render json: result.error, status: :not_found
+    end
+  end
```

Comment:
I tend to prefer to maintain consistency and to keep using interactors throughout the app, even in simple cases like this. That's just my preference thought :smile:. You can argue that abstracting something so simple into an interactor could make it harder to read.

It also depends on what you are doing. For some apps, `show` could required a more complex flow that you might want to check what the result was. Rails automatically handles `not_found` for ActiveRecord and Postgres, but if you decide to use something else then you have to catch those errors yourself. The interactor patter helps with that.

I think it really boils down to a matter of preference. Rails has conventions for things and some people like just following the same conventions. It makes it easier to read and understand an app if you know where to look. :smiley: